yasa-solr-plugin/src/main/java/io/github/kezhenxu94/YasaHandler.java

```diff
@@ -36,35 +42,49 @@
 import org.apache.solr.response.SolrQueryResponse;
 import org.apache.solr.security.PermissionNameProvider;
 
-import static io.github.kezhenxu94.YasaHandler.PLUGIN_PATH;
-
 @EndPoint(
   method = METHOD.GET,
-  path = PLUGIN_PATH + "/*",
+  path = "$path-prefix/*",
   permission = PermissionNameProvider.Name.CONFIG_READ_PERM
 )
 @RequiredArgsConstructor
-@SuppressWarnings("unused")
-public class YasaHandler {
-  public static final String PLUGIN_PATH = "/plugin/yasa";
-
+public class YasaHandler implements ResourceLoaderAware {
+  @SuppressWarnings("unused")
   private final CoreContainer coreContainer;
 
+  private ResourceLoader loader = null;
+
+  @Override
+  public void inform(ResourceLoader loader) throws IOException {
+    this.loader = loader;		
+  }
+
   @Command
   public void call(SolrQueryRequest req, SolrQueryResponse rsp) throws IOException {
-    final String path = req.getHttpSolrCall().getPath();
+    String path = req.getHttpSolrCall().getPath();
+	String filepath = resolveFilePath(path);
 
-    String filepath = path.substring(path.indexOf(PLUGIN_PATH) + PLUGIN_PATH.length());
-    if ("".equalsIgnoreCase(filepath) || "/".equalsIgnoreCase(filepath)) {
-      filepath = "/index.html";
-    }
+	InputStream in = loader.openResource(filepath);
+	if (in == null) {
+		throw new SolrException(ErrorCode.NOT_FOUND, "File not found: " + filepath);
+	}
 
+    final byte[] data;
+    final String contentType;
+    
+    if ("".equals(filepath)) {
+      String indexPath = path.endsWith("/")? path + "index.html": path + "/index.html";
+      indexPath = indexPath.replaceAll("\\_\\_\\_\\_v2", "v2");
+      data = ("<meta http-equiv=\"Refresh\" content=\"0; url='" + indexPath + "'\" />").getBytes(Charset.forName("UTF-8"));
+      contentType = ContentType.TEXT_HTML.getMimeType();
+    } else {
+      data = IOUtils.toByteArray(in);
```

Comment:
I meant 81

Reply:
Content Refresh as a return body seems weird to me. I would expect HTTP Status. Can you explain this to me?

Reply:
Solr doesn't support returning arbitrary return codes (like 304). This was the only way I could think of.
What is weird about this?

Reply:
most APIs return status codes. This return value of content refresh is not the first time I've seen something like this, but it's been many years. 

I don't know if redirect status codes are arbitrary if you are redirecting.

Reply:
Feel free to update the PR if you find the best alternative here. I'll probably have no further time to spend on this until after Solr 8.6 is released, since I have a ton of work to finish for that release.